src/framework/mpas_stream_manager.F

```diff
@@ -4845,6 +4964,51 @@ logical function is_decomposed_dim(dimName) !{{{
 
     end function is_decomposed_dim !}}}
 
+    
+    
+    integer function global_dim_size(block, dimNames, dimSizes, nDim) !{{{
+    use iso_c_binding, only : c_sizeof
+
+    implicit none
+
+    character(len=*), intent(in), dimension(:) :: dimNames
+    integer, intent(in), dimension(:) :: dimSizes
+    type(block_type), intent(in) :: block
+    integer, intent(in) :: nDim
+    integer, pointer :: block_dim_size
+    integer :: iDim, sum_block_dim_size
+    character(len=StrKIND):: message
+
+    global_dim_size = 1
+    do iDim = 1, nDim
```

Comment:
It could save passing in the `nDim` argument to just use `size(dimSizes)`.

Reply:
Done